### DISCUSSION TITLE
Using luarocks

### DISCUSSION THREAD

**Discussion by pogo5422** [wbthomason/packer.nvim#173] [Q&A] [answered]
I want to write a few things inside my neovim's config that depend on luarocks libraries. How should I install them w/ packer? Currently, I've just inserted `rocks` key in some random `use` statement so it looks like
```lua
use {'wbthomason/packer.nvim', opt = true, rocks = {'lua-cjson', 'httpclient'}}
```
, but I wonder if there is a better way, i.e. `use {rocks = {...}}`, without plugin name

**Comment by wbthomason** [comment, marked as answer]
Right now, that's unfortunately the best way. You can use `PackerRocks install {your rocks}`, but if you're using `packer` as your regular plugin manager, then those rocks will be deleted on the next clean/sync (the `PackerRocks` command is designed for people who want to manage Luarocks dependencies without using `packer` as a plugin manager).

I think, later today, I may add a function `use_rocks` that just takes a list of rocks to add. This would make your use case much more straightforward.

**Reply by wbthomason** [reply]
I have now added `packer.use_rocks` for this purpose. You can now write something like
```lua
local use_rocks = packer.use_rocks
use_rocks {'lua-cjson', 'httpclient'}
```

**Reply by akinsho** [reply]
This is a very, *very* minor point but I wonder if `use_rocks` could be supplied to `startup` in the same way so a user can do
```lua
require('packer').startup(function(use, use_rocks)
  -- 
end)
```
It's really not a blocker of any sort though

**Comment by wbthomason** [comment]
Fair point/good idea. Done in 3a8543e.
